request_a_govuk_domain/request/views.py

```diff
@@ -384,24 +384,28 @@ def post(self, request):
         :return:
         """
         reference_ = request.session.get(APPLICATION_REFERENCE)
-        self.save_application_to_database_and_send_confirmation_email(
+        status = self.save_application_to_database_and_send_confirmation_email(
```

Comment:
You can have a try catch statement here and get rid of the boolean checking? This is because DB will throw integrity exception when there is a ref conflict, the above check for the existance of the application may not work also if both the applications are created at the same time very fast as both will go through the if condition..